### ISSUE TITLE
Switch to BrowserSync

### ISSUE THREAD

**Issue by sindresorhus** [yeoman/generator-webapp_DEPRECATED#400]
[BrowserSync](http://www.browsersync.io/) is pretty good and supports multiple devices and lot more. There's a [grunt task](https://github.com/shakyshane/grunt-browser-sync).

**Comment by markthethomas** [comment]
:+1:

**Comment by KingScooty** [comment]
:+1:

**Comment by eddiemonge** [comment]
What does it give us over livereload?

**Comment by sindresorhus** [comment]
> supports multiple devices and lot more

**Comment by htanjo** [comment]
+1

**Comment by stevemao** [comment]
Hi @sindresorhus , would you like to completely ditch grunt-contrib-connect or just for `serve`?

**Comment by sindresorhus** [comment]
@stevemao Ditch it. Pretty sure BrowserSync can do anything grunt-contrib-connect can and more. If there's anything missing in BrowserSync open an issue there. The author is very responsive :)

**Comment by diagramatics** [comment]
Tried converting to BrowserSync on my app, and apparently the only problem I've hit on converting is the fact that compiling from SASS triggers it once, then autoprefixer triggers it again. Apart from that, everything's working well.

This has been reported in shakyShane/grunt-browser-sync#50. Unless you guys are okay with events and accessing the API, we can't convert it just yet. I can share my Gruntfile.js if needed.

**Comment by stevemao** [comment]
@diagramatics true...

**Comment by addyosmani** [comment]
Agree we should switch to BrowserSync. We already use this successfully in Web Starter Kit and haven't really run into issues with Sass and Autoprefixer working well in live-reload together. That said, it doesn't look like there's been much movement on the reported issue above.

Has anyone else been looking into using grunt-browser-sync with this generator lately?

**Comment by shakyShane** [comment]
on board, will send pr

**Comment by shakyShane** [comment]
Arg! I jumped in a little early there - it seems like this would suffer from the same issue as described here: http://www.shakyshane.com/javascript/nodejs/browser-sync/2014/08/24/browser-sync-plus-grunt/

There's a reason live-reload was baked into the watch plugin... :(

Maybe this stays on hold until someone has a better idea of how achieve this (assuming you don't want to do what's in my post above, which is actually how we use BrowserSync at my day-job)

tl;dr - something like this is needed https://gist.github.com/shakyShane/0edcffd9e2c0c9e33dd5

**Comment by shakyShane** [comment]
For future reference, it was trivial to swap out connect/livereload for BrowserSync. https://gist.github.com/shakyShane/3ef441413d7726168c4d

**Comment by brunowego** [comment]
+1
